src/Components/RightSidebar/RightSidebar.jsx

```diff
@@ -30,20 +35,30 @@ export const RightSidebar = () => {
                       <h1>{user.firstName}</h1>
                       <h3>@{user.username}</h3>
                     </div>
-                    <button className={styles.follow_btn}>Unfollow</button>
+                    <button
+                      onClick={() => unfollowUser(user._id)}
+                      className={styles.follow_btn}
+                    >
+                      Unfollow
+                    </button>
                   </div>
                 </div>
               </div>
             );
-          })
-        ) : (
-          <h1>No Following</h1>
-        )}
+          })}
       </div>
       <div className={styles.follower_section}>
+        
         <h1>Who to Follow</h1>
         {users
           .filter((user) => user.username !== userState.username)
+          .filter((user) =>
+            userFollowing.length === 0
+              ? user
+              : !userFollowing
+                  .map((user) => user.username)
+                  .includes(user.username)
+          )
```

Comment:
suggestion: this can be a hook or a utility function to reduce too much of logic in the component and  achieve more re-usability

Reply:
Thanks for the suggestion will check this out